cinema/views.py

```diff
@@ -116,14 +142,20 @@
 class OrderPagination(PageNumberPagination):
     page_size = 10
     max_page_size = 100
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet,
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
     def get_queryset(self):
         return Order.objects.filter(user=self.request.user)
```

Comment:
The `get_queryset` method in the `OrderViewSet` class is not considering the case when the user is not authenticated. This can lead to a 500 error if an anonymous user tries to access this endpoint. You need to handle this case and return an appropriate response.